### PR TITLE
Initial Losses API Template

### PR DESCRIPTION
**Pull request by hassiahk**
Hello Together,

As discussed in https://github.com/Quick-AI/quickvision/issues/43, created an initial template for Losses API. Let me know if this good enough.

### CODE REVIEW

**Comment by oke-aditya** [comment]
@hassiahk  Template looks good, I will thoroughly review it once more though. Ensure that we don't fail DETR Tests where we use this loss.

File: quickvision/losses/functional/dice_loss.py [hunk 1]
```diff
@@ -1,21 +1,33 @@
 import torch
+from typing import Optional
 
 __all__ = ["dice_loss"]
 
 
-def dice_loss(inputs, targets):
+def dice_loss(
+    inputs, targets, weights: Optional[torch.Tensor] = None, reduction: str = "mean"
```
Comment (oke-aditya) [inline quickvision/losses/functional/dice_loss.py:8]: can this be `sum` as default.

**Review by oke-aditya — COMMENTED** [review]
Can you add tests for dice loss as well ? Or we can leave in new PR

File: quickvision/losses/functional/dice_loss.py (hunk 1 above)
Comment (hassiahk) [reply quickvision/losses/functional/dice_loss.py]: Done!

**Review by oke-aditya — APPROVED** [review]
LGTM !
